gramineproject/gramine
# [Meson] Update musl to v1.2.4

## Description of the changes 

This update is important because programs built against newer musl may crash under older musl (e.g. v1.2.2 which Gramine had previously). In particular, Gramine segfaulted on Alpine 3.18.0 on `/bin/sh` (resolves to Busybox) apps, because Busybox is built against musl v1.2.4 and therefore requires the new relocation type `RELR`.

## How to test this PR? 

CI should be enough.

Try to run some apps on Alpine 3.18.0.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/1393)
